### ISSUE TITLE
MacType service mode imcompatible with KMS_VL_ALL series

### ISSUE THREAD

**Issue by NathanIceSea** [snowie2000/mactype#654]
As title suggests, as long as mactype runs in Service mode or Registry mode, KMS_VL_ALL activate shall fail, it remains unknown what cause the problem, maybe it has something to do with the SppExtComObjHook.dll or womething

By the way, the original mactype 2013 with patch 26 does not have this problem, however that version does not support unload EasyHK64.dll which violates with Steinberg eLicenser the USB Licenser, feels so sad

**Comment by snowie2000** [comment]
I really don't have any idea how this thing works, so I have no idea how to debug it...
Does it provide any sort of logs?

**Comment by ghost** [comment]
> As title suggests, as long as mactype runs in Service mode or Registry mode, KMS_VL_ALL activate shall fail, it remains unknown what cause the problem, maybe it has something to do with the SppExtComObjHook.dll or womething
> 
> By the way, the original mactype 2013 with patch 26 does not have this problem, however that version does not support unload EasyHK64.dll which violates with Steinberg eLicenser the USB Licenser, feels so sad



> I really don't have any idea how this thing works, so I have no idea how to debug it...
> Does it provide any sort of logs?

Add SppExtComObj.Exe to exclusion list (UnloadDLL list) in MacType.ini ,the problem will solve.

**Comment by snowie2000** [comment]
Can you activate your softwares with normal KMS servers? If yes, there must be conflicts between `SppExtComObjHook.dll` and mactype.

**Comment by ghost** [comment]
> Can you activate your softwares with normal KMS servers? If yes, there must be conflicts between `SppExtComObjHook.dll` and mactype.

Yes, normal KMS servers works well, and crash only happens when MacType is loaded by using registry mode. An eventlog will be recorded that shows SppExtComObj.Exe is crashed. The `SppExtComObjHook.dll` comes with KMS_VL_ALL or HEU KMS 19.6+ when auto renewal mode is enabled.
 
crashlog here:
Faulting application name: SppExtComObj.exe, version: 10.0.18362.719, time stamp: 0xa180932c
Faulting module name: unknown, version: 0.0.0.0, time stamp: 0x00000000
Exception code: 0xc0000005
Fault offset: 0x00000185703c0036
Faulting process id: 0x108c
Faulting application start time: 0x01d624e733917883
Faulting application path: C:\Windows\system32\SppExtComObj.exe
Faulting module path: unknown
Report Id: bea68524-9925-443e-adfa-27cde14627da
Faulting package full name:
